From f88a63f67c6ca14f5bb8452149da71a4c9d56a80 Mon Sep 17 00:00:00 2001
From: Robin Lee <cheeselee@fedoraproject.org>
Date: Thu, 13 Jun 2019 18:12:17 +0800
Subject: [PATCH] udevadm set as ExecStartPre to make sure interface-rename
 always run

If interface-rename was not run during system booting, netif names will left in bad format
and the system is unusable. This change makes udevadm run with "ExecStartpre=-" so that
failure of udevadm will not prevent interface-rename from starting.

Signed-off-by: Robin Lee <cheeselee@fedoraproject.org>
---
 systemd/interface-rename.service | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/systemd/interface-rename.service b/systemd/interface-rename.service
index f556138..0d2c997 100644
--- a/systemd/interface-rename.service
+++ b/systemd/interface-rename.service
@@ -8,9 +8,9 @@ Before=xcp-networkd.service network.service
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/usr/bin/udevadm settle --timeout=180
-ExecStart=/usr/bin/udevadm trigger
-ExecStart=/usr/bin/udevadm settle --timeout=180
+ExecStartPre=-/usr/bin/udevadm settle --timeout=180
+ExecStartPre=-/usr/bin/udevadm trigger
+ExecStartPre=-/usr/bin/udevadm settle --timeout=180
 ExecStart=/usr/bin/python /etc/sysconfig/network-scripts/interface-rename.py --rename
 
 [Install]
-- 
2.21.0

